File: benchmark/go/users_api.go
package main

import (
	"encoding/json"
	"net/http"
)

// UsersAPI is API implementation of /users root endpoint
type UsersAPI struct {
}

// Get is the handler for GET /users
// Get random user
func (api UsersAPI) Get(w http.ResponseWriter, r *http.Request) {
	respBody := User{Name: "John", Username: "Doe"}
	json.NewEncoder(w).Encode(&respBody)
	// uncomment below line to add header
	// w.Header().Set("key","value")
}

// Post is the handler for POST /users
// Add user
func (api UsersAPI) Post(w http.ResponseWriter, r *http.Request) {
	var reqBody User

	// decode request
	if err := json.NewDecoder(r.Body).Decode(&reqBody); err != nil {
		w.WriteHeader(400)
		return
	}

	// validate request
	/*if err := reqBody.Validate(); err != nil {
		w.WriteHeader(400)
		w.Write([]byte(`{"error":"` + err.Error() + `"}`))
		return
	}*/
	respBody := reqBody.Name
	json.NewEncoder(w).Encode(&respBody)
	// uncomment below line to add header
	// w.Header().Set("key","value")
}

// usernameGet is the handler for GET /users/{username}
// Get information on a specific user
func (api UsersAPI) usernameGet(w http.ResponseWriter, r *http.Request) {
	var respBody User
	json.NewEncoder(w).Encode(&respBody)
	// uncomment below line to add header
	// w.Header().Set("key","value")
}

File: benchmark/nim/users_api.nim
import jester, marshal

import User


proc usersGet*(req: Request) : tuple[code: HttpCode, content: User] =
  # Get random user
  
  var respBody: User
  respBody = User(name:"John", username: "Doe")
  result = (code: Http200, content: respBody)

proc usersPost*(req: Request) : tuple[code: HttpCode, content: string] =
  # Add user
  let reqBody = to[User](req.body)
  var respBody: string
  respBody = reqBody.name
  result = (code: Http200, content: respBody)

proc usersByUsernameGet*(username: string, req: Request) : tuple[code: HttpCode, content: User] =
  # Get information on a specific user
  
  var respBody: User
  result = (code: Http200, content: respBody)

